frontend/src/components/activePage/TableInstance/TableInstanceLogic.tsx

```diff
@@ -0,0 +1,168 @@
+import { FetchPolicy } from '@apollo/client';
+import { FC, useState, useEffect, useContext } from 'react';
+import { notification, Spin } from 'antd';
+import Button from 'antd-button-color';
+import { Instance, WorkspaceRole } from '../../../utils';
+import './TableInstance.less';
+import TableInstance from './TableInstance';
+import { AuthContext } from '../../../contexts/AuthContext';
+import {
+  useDeleteInstanceMutation,
+  useOwnedInstancesQuery,
+  UpdatedOwnedInstancesSubscriptionResult,
+  OwnedInstancesQuery,
+} from '../../../generated-types';
+import { updatedOwnedInstances } from '../../../graphql-components/subscription';
+
+export interface ITableInstanceLogicProps {
+  viewMode: WorkspaceRole;
+  showGuiIcon: boolean;
+  extended: boolean;
+  tenantNamespace: string;
+}
+
+const fetchPolicy_networkOnly: FetchPolicy = 'network-only';
+
+const TableInstanceLogic: FC<ITableInstanceLogicProps> = ({ ...props }) => {
+  const { tenantNamespace, viewMode, extended, showGuiIcon } = props;
+  const { userId } = useContext(AuthContext);
+  const [dataInstances, setDataInstances] = useState<OwnedInstancesQuery>();
+  const [deleteInstanceMutation] = useDeleteInstanceMutation();
+
+  const startInstance = (idInstance: string, idTemplate: string) => {};
+  const stopInstance = (idInstance: string, idTemplate: string) => {};
+
+  const {
+    loading: loadingInstances,
+    error: errorInstances,
+    subscribeToMore: subscribeToMoreInstances,
+  } = useOwnedInstancesQuery({
+    variables: { tenantNamespace },
+    onCompleted: setDataInstances,
+    fetchPolicy: fetchPolicy_networkOnly,
+  });
+
+  useEffect(() => {
+    if (!loadingInstances) {
+      subscribeToMoreInstances({
+        document: updatedOwnedInstances,
+        variables: { tenantNamespace },
+        updateQuery: (prev, { subscriptionData }) => {
+          const {
+            data,
+          } = subscriptionData as UpdatedOwnedInstancesSubscriptionResult;
+
```

Comment:
Isn't this logic quite similar to the one which @cheina97 made? Can't we find a way to recycle it? Or at least parts of the code in this hook?

Reply:
we could try to do it in the other PR where we clean and organize the code removing also duplicates